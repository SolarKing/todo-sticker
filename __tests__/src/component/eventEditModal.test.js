// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import EditModal from '../../../src/component/eventEditModal';
import { Modal, DatePicker, TimePicker, Input } from 'antd';
import moment from 'moment';

describe('EditModal', () => {
    let wrapper: ShallowWrapper;
    let handleAddEvent = jest.fn();
    let onSave = jest.fn();
    
    beforeAll(() => {
        jest.spyOn(Date.prototype, 'getTime').mockReturnValueOnce(0);
        wrapper = shallow(
            <EditModal
                visible={true}
                title='Test'
                title='Title'
                todoId='1'
                defaultDueDatetime={0}
                defaultDescription='Test description'
                handleAddEvent={handleAddEvent}
                onSave={onSave}
            />
        );
        
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle date changed', () => {
        let changeExpect = moment('2019-01-10T10:45:20+00:00');
        wrapper.find(DatePicker).simulate('change', changeExpect);
        let result = new Date(wrapper.state().dueDatetime);
        expect(result.getUTCFullYear()).toEqual(2019);
        expect(result.getUTCMonth() + 1).toEqual(1);
        expect(result.getUTCDate()).toEqual(10);
    });

    it('should handle time changed', () => {
        let changeExpect = moment('2019-01-10T10:45:20+00:00');
        wrapper.find(TimePicker).simulate('change', changeExpect);
        let result = new Date(wrapper.state().dueDatetime);
        expect(result.getUTCHours()).toEqual(10);
        expect(result.getUTCMinutes()).toEqual(45);
    });

    it('should handle descriptions changed', () => {
        const event = {
            currentTarget: { 
                value: 'Hello there!' 
            }
        };
        wrapper.find(Input.TextArea).simulate('change', event);
        expect(wrapper.state().desctiption).toEqual('Hello there!');
    });

    it('should handle ok', () => {
        let changeExpect = moment('2019-01-10T10:45:20+00:00');
        const event = {
            currentTarget: { 
                value: 'Hello!' 
            }
        };

        wrapper.find(DatePicker).simulate('change', changeExpect);
        wrapper.find(TimePicker).simulate('change', changeExpect);
        wrapper.find(Input.TextArea).simulate('change', event);
        wrapper.find(Modal).simulate('ok');

        expect(handleAddEvent.mock.calls.length).toEqual(1);
        expect(handleAddEvent.mock.calls[0][0]).toEqual({
            description: 'Hello!',
            dueDatetime: moment('2019-01-10T10:45:00+00:00').toDate().getTime(),
        });
    });

});