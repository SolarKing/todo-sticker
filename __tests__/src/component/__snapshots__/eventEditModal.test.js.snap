// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditModal should render correct 1`] = `
<Modal
  confirmLoading={false}
  defaultDescriptin=""
  destroyOnClose={true}
  handleAddEvent={[MockFunction]}
  maskTransitionName="fade"
  okType="primary"
  onOk={[Function]}
  onSave={[MockFunction]}
  prefixCls="ant-modal"
  title="Title"
  todoId="1"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <p>
    Due date
  </p>
  <PickerWrapper
    allowClear={false}
    defaultValue={"1970-01-01T00:00:00.000Z"}
    format="YYYY-MM-DD"
    inputPrefixCls="ant-input"
    locale={Object {}}
    onChange={[Function]}
    onOk={[Function]}
    onOpenChange={[Function]}
    popupStyle={Object {}}
    prefixCls="ant-calendar"
    transitionName="slide-up"
  />
  Â 
  <TimePicker
    align={
      Object {
        "offset": Array [
          0,
          -2,
        ],
      }
    }
    allowEmpty={false}
    defaultValue={"1970-01-01T00:00:00.000Z"}
    disabled={false}
    focusOnOpen={true}
    format="HH:mm"
    hideDisabledOptions={false}
    onChange={[Function]}
    placement="bottomLeft"
    prefixCls="ant-time-picker"
    transitionName="slide-up"
  />
  <p>
    eventDescription
  </p>
  <TextArea
    defaultValue="Test description"
    onChange={[Function]}
    prefixCls="ant-input"
    rows={4}
  />
</Modal>
`;
